addons/udes_security/models/res_archiving_restrictions.py

```diff
@@ -0,0 +1,82 @@
+from odoo import fields, models, _, SUPERUSER_ID
+from odoo.exceptions import ValidationError
+
+
+class ResArchivingRestrictions(models.Model):
+    """
+    Configurable access to _only_ record archiving & unarchiving
+
+    If an Archiving Restriction is created for a model, the group_ids set will determine
+    which groups have the facility to archive and unarchive records on the model.
+
+    This includes backend writes to the `active` field, as well as hiding
+    the Archive/Unarchive action from the model, and attempting to remove active
+    fields & smart buttons from form views.
+    """
+
+    _name = "res.archiving.restrictions"
+    _description = "Record Archiving Restrictions"
+    _sql_constraints = [
+        (
+            "model_id_uniq",
+            "unique(model_id)",
+            _(
+                "An Archiving Restriction already exists for this model. Please edit the existing one"
+            ),
+        )
+    ]
+
+    model_id = fields.Many2one(
+        "ir.model", string="Model to apply restriction to", required=True, index=True
+    )
+    name = fields.Char(string="Model Name", related="model_id.model", readonly=True)
+    group_ids = fields.Many2many(
+        "res.groups",
+        string="Groups permitted to archive & unarchive",
+        index=True,
+        help="""
+        If no groups are configured, ALL users will be prevented from archiving (except superuser)
+        If the user is in any of the configured groups, they will be permitted access
+        If no restriction on a model is intended, then do not create a restriction record for that model
+        """,
+    )
+
+    def _get_can_archive(self, model_name):
+        """
+        Test against the model to see if our user is
```

Comment:
Suggestion: `Check against the model to see if active user is permitted to archive/unarchive the record.`

"Test" and "our user" read a bit odd to me.